packages/core/src/components/segmented-control/segmentedControl.tsx

```diff
@@ -104,14 +108,48 @@ export const SegmentedControl: React.FC<SegmentedControlProps> = React.forwardRe
         [onValueChange],
     );
 
+    const handleKeyDown = React.useCallback(
+        (e: React.KeyboardEvent<HTMLDivElement>) => {
+            const direction = getArrowKeyDirection(e, true);
+            const { current: outerElement } = outerRef;
+            if (direction == undefined || !outerElement) return;
+
+            const focusedElement = Utils.getActiveElement(outerElement)?.closest<HTMLButtonElement>("button");
+            if (!focusedElement) return;
+
+            // must rely on DOM state because we have no way of mapping `focusedElement` to a React.JSX.Element
+            const enabledOptionElements = Array.from(outerElement.querySelectorAll("button")).filter(
```

Comment:
Could we leverage selector query to filter out disabled buttons? e.g.
```suggestion
            const enabledOptionElements = Array.from(outerElement.querySelectorAll("button:not(:disabled)"));
```

Reply:
Good idea, done!